realm-transformer/src/main/kotlin/io/realm/transformer/RealmOptionalAPITransformer.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.transformer
+
+import com.android.build.api.transform.Format
+import com.android.build.api.transform.Context
+import com.android.build.api.transform.Transform
+import com.android.build.api.transform.QualifiedContent.ContentType
+import com.android.build.api.transform.QualifiedContent.Scope
+import com.android.build.api.transform.QualifiedContent.DefaultContentType
+import com.android.build.api.transform.TransformInput
+import com.android.build.api.transform.TransformOutputProvider
+import com.google.common.collect.ImmutableSet
+import io.realm.annotations.internal.OptionalAPI
+import org.slf4j.LoggerFactory
+import java.util.HashSet
+
+class RealmOptionalAPITransformer : Transform() {
+
+    private val logger = LoggerFactory.getLogger("realm-logger")
+
+    override fun getName(): String? = "RealmOptionalAPITransformer"
+
+    override fun isIncremental(): Boolean = false
+
+    override fun getInputTypes(): MutableSet<ContentType>? = ImmutableSet.of(DefaultContentType.CLASSES)
+
+    override fun getScopes(): MutableSet<Scope>? = ImmutableSet.of(Scope.EXTERNAL_LIBRARIES)
+
+    override fun getReferencedScopes(): MutableSet<Scope>? = ImmutableSet.of(Scope.PROJECT, Scope.PROJECT_LOCAL_DEPS,
+            Scope.SUB_PROJECTS, Scope.SUB_PROJECTS_LOCAL_DEPS, Scope.EXTERNAL_LIBRARIES)
+
+    override fun transform(context: Context?,
+                           inputs: MutableCollection<TransformInput>?,
+                           referencedInputs: MutableCollection<TransformInput>?,
+                           outputProvider: TransformOutputProvider?,
+                           isIncremental: Boolean) {
+
+        val classNames = HashSet<String>()
+        inputs!!.appendAllToClassNames(classNames)
```

Comment:
Ah yes sorry, I think the method name confused me. `appendAll` would normally apply to `inputs`, but in this case it actually the other way around. Would it make more sense to put the extension method on the HashSet? Or find a better name ( I don't have anything better though).